app/models.py

```diff
@@ -888,14 +888,14 @@ def apply(self, domain, post_records):
 
         # Adjustment to add multiple records which described in https://github.com/ngoduykhanh/PowerDNS-Admin/issues/5#issuecomment-181637576
         final_records = []
-        if NEW_SCHEMA:
-            records = sorted(records, key = lambda item: (item["name"], item["type"]))
-            for key, group in itertools.groupby(records, lambda item: (item["name"], item["type"])):
+        records = sorted(records, key = lambda item: (item["name"], item["type"]))
+        for key, group in itertools.groupby(records, lambda item: (item["name"], item["type"], item["changetype"], item["ttl"])):
```

Comment:
Yes, you are correct. I actually had the same issues (one of them would be deleted), but upon more tests, it didn't happen. I think I might've tested that last part on the wrong branch.